app/main.py

```diff
@@ -2,24 +2,76 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: str, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: str, owner: str) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: str, value: str) -> None:
```

Comment:
The `__set__` method's `instance` parameter should be of type `object`, not `str`. This method is called with the instance.